apps/dashboard/app/models/script.rb

```diff
@@ -153,6 +162,18 @@ def most_recent_job_id
     most_recent_job['id']
   end
 
+  def write_job_options_to_cache(opts)
+    File.write(cache_file_path, opts.to_json)
+  end
+
+  def cache_file_path
+    Pathname.new(project_dir).join(".ondemand/scripts/#{id}_opts.json")
```

Comment:
I think you should use `Script.scripts_dir` here and maybe rename the file to show that it's a cache?

```suggestion
    Pathname.new("#{Script.scripts_dir(project_dir)}/#{id}_cache.json")
```

Reply:
Renaming makes sense. 

When I try to use the `Script.scripts_dir(project_dir)` I get errors from the `show` view about:
```
undefined method `most_recent_job_id' for nil:NilClass
```
I'll play with it some more since I'd prefer to not hardcode the path like i have.

Reply:
Hmmm. You may have found a bug.

Reply:
Ok.  I played with it for a bit but am unsure why it's happening. For now I'm going to leave the commented code and use what I have, but open an issue for the bug to be looked at separately and change that when we figure it out. 

Unless you think it makes sense to try and debug that here, happy to do that instead.

Reply:
Sure that's fine, just add a `TODO` comment near it to increase the likelihood we circle back to it.

Reply:
Cool. I opened an issue here for this #2770